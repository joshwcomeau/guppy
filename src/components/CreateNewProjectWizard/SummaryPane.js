// @flow
import React, { PureComponent, Fragment } from 'react';
import styled from 'styled-components';

import { RAW_COLORS, COLORS } from '../../constants';

import Paragraph from '../Paragraph';
import FadeIn from '../FadeIn';
import Spacer from '../Spacer';
import Logo from '../Logo';
import ExternalLink from '../ExternalLink';

import ImportExisting from './ImportExisting';

import type { ProjectType } from '../../types';
import type { Field, Step } from './types';

type Props = {
  currentStep: Step,
  activeField: ?Field,
  projectType: ?ProjectType,
};

class SummaryPane extends PureComponent<Props> {
  renderPaneContents() {
    const { currentStep, activeField, projectType } = this.props;

    // If we're still in the first step, we want to show our intro details.
    if (currentStep === 'projectName') {
      return (
        <IntroWrapper>
          <FadeIn key="intro-t">
            <Logo size="large" />

            <Spacer size={30} />
            <StepTitle>Create new project</StepTitle>
            <Paragraph>
              Let's start by giving your new project a name.
            </Paragraph>

            <Spacer size={130} />
            <ImportExisting />
          </FadeIn>
        </IntroWrapper>
      );
    }

    // After that first step, there's a "default" display for each step,
    // but that can be overridden with active focus.
    const focusField = activeField || currentStep;

    switch (focusField) {
      case 'projectName': {
        return (
          <Fragment>
            <FadeIn key="s1-1">
              <StepTitle>Project Name</StepTitle>
              <Paragraph>
                Don't stress too much about your project's name! You can always
                change this later.
              </Paragraph>
            </FadeIn>
          </Fragment>
        );
      }

      case 'projectType': {
        let details;

        switch (projectType) {
          default: {
            details = (
              <Paragraph>
                Guppy can create projects of different types. Click a type to
                learn more about it.
              </Paragraph>
            );
            break;
          }
          case 'create-react-app': {
            details = (
              <Fragment>
                <Paragraph>
                  <strong>Vanilla React</strong>
                </Paragraph>
                <Paragraph>
                  Vanilla React projects use create-react-app, an official
                  command-line tool built by Facebook for bootstrapping React
                  applications.
                </Paragraph>
                <Paragraph>
                  It's a fantastic general-purpose tool, and is the recommended
                  approach if you're looking to become a skilled React
                  developer.
                </Paragraph>
                <Paragraph>
                  <ExternalLink
                    color={COLORS.textOnBackground}
                    hoverColor={COLORS.textOnBackground}
                    href="https://github.com/facebook/create-react-app"
                  >
                    <strong>Learn more about create-react-app.</strong>
                  </ExternalLink>
                </Paragraph>
              </Fragment>
            );
            break;
          }
          case 'gatsby': {
            details = (
              <Fragment>
                <Paragraph>
                  <strong>Gatsby</strong>
                </Paragraph>
                <Paragraph>
                  Gatsby is a blazing fast static site generator for React.
                </Paragraph>
                <Paragraph>
                  It's great for building blogs and personal websites, and
                  provides amazing performance out-of-the-box. A great choice
                  for quickly getting products built.
                </Paragraph>
                <Paragraph>
                  <ExternalLink
                    color={COLORS.textOnBackground}
                    hoverColor={COLORS.textOnBackground}
                    href="https://www.gatsbyjs.org/"
                  >
                    <strong>Learn more about Gatsby.</strong>
                  </ExternalLink>
                </Paragraph>
              </Fragment>
            );
            break;
          }
          case 'nextjs': {
            details = (
              <Fragment>
                <Paragraph>
                  <strong>Next.js</strong>
                </Paragraph>
                <Paragraph>
                  Next.js is a lightweight framework for static and
                  server-rendered applications.
                </Paragraph>
                <Paragraph>
                  Server-rendered by default. No need to worry about routing. A
                  great choice for quickly getting products built with
                  server-side rendering by a Node.js server.
                </Paragraph>
                <Paragraph>
                  <ExternalLink
                    color={COLORS.textOnBackground}
                    hoverColor={COLORS.textOnBackground}
                    href="https://nextjs.org/learn/"
                  >
                    <strong>Learn more about Next.js.</strong>
                  </ExternalLink>
                </Paragraph>
              </Fragment>
            );
            break;
          }
        }
        return (
          <Fragment>
            <FadeIn key="s2t">
              <StepTitle>Project Type</StepTitle>
              {details}
            </FadeIn>
          </Fragment>
        );
      }

      case 'projectIcon': {
        return (
          <Fragment>
            <FadeIn key="s3t">
              <StepTitle>Project Icon</StepTitle>

              <Paragraph>
                Choose an icon, to help you recognize this project from a list.
              </Paragraph>
            </FadeIn>
          </Fragment>
        );
      }
      case 'projectStarter': {
        // todo: why is a key needed on FadeIn? Was s3t.
        // todo: should we rename projectStarter to be mores specific as this is Gatsby only.
        return (
          <Fragment>
            <FadeIn>
              <StepTitle>Gatsby Starter</StepTitle>

              <Paragraph>
                Please enter a starter for your project (e.g.
                gatsby-starter-blog or repo. url) or pick one from the starters
                list.
              </Paragraph>
              <Paragraph>
                This step is optional. Just leave the field empty to use the
                default Gatsby starter. But picking a starter will help to
                bootstrap your project e.g. you can easily create your own blog
                by picking one of the blog starter templates.
              </Paragraph>
              <Paragraph>
                For a better overview you can also have a look at the{' '}
                <ExternalLink
                  color={RAW_COLORS.white}
                  hoverColor={RAW_COLORS.white}
                  href="https://www.gatsbyjs.org/starters/"
                >
                  Gatsby starters library
                </ExternalLink>
              </Paragraph>
            </FadeIn>
          </Fragment>
        );
      }

      default:
        throw new Error('Unrecognized `focusField`: ' + focusField);
    }
  }

  render() {
    return <Wrapper>{this.renderPaneContents()}</Wrapper>;
  }
}

const Wrapper = styled.div`
  text-shadow: 1px 1px 0px rgba(13, 37, 170, 0.1);
`;

const StepTitle = styled.h1`
  font-size: 28px;
  margin-bottom: 30px;
`;

const IntroWrapper = styled.div`
  text-align: center;
  padding-top: 20px;
`;

export default SummaryPane;
