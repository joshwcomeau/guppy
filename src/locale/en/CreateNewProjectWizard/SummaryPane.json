{
  "createNewProjectWizard.summaryPane.start.title": "Create new project",
  "createNewProjectWizard.summaryPane.start.content": "Let's start by giving your new project a name.",
  "createNewProjectWizard.summaryPane.projectName.title": "Project Name",
  "createNewProjectWizard.summaryPane.projectName.content": "Don't stress too much about your project's name! You can always change this later.",
  "createNewProjectWizard.summaryPane.projectType.title": "Project Type",
  "createNewProjectWizard.summaryPane.projectType.content": "Guppy can create projects of different types. Click a type to learn more about it.",
  "createNewProjectWizard.summaryPane.projectType.reactIntro": "Vanilla React projects use create-react-app, an official command-line tool built by Facebook for bootstrapping React applications.",
  "createNewProjectWizard.summaryPane.projectType.reactComment": "It's a fantastic general-purpose tool, and is the recommended approach if you're looking to become a skilled React developer.",
  "createNewProjectWizard.summaryPane.projectType.reactLearn": "Learn more about create-react-app.",
  "createNewProjectWizard.summaryPane.projectType.gatsbyIntro": "Gatsby is a blazing fast static site generator for React.",
  "createNewProjectWizard.summaryPane.projectType.gatsbyComment": "It's great for building blogs and personal websites, and provides amazing performance out-of-the-box. A great choice for quickly getting products built.",
  "createNewProjectWizard.summaryPane.projectType.gatsbyLearn": "Learn more about Gatsby.",
  "createNewProjectWizard.summaryPane.projectType.nextjsIntro": "Next.js is a lightweight framework for static and server-rendered applications.",
  "createNewProjectWizard.summaryPane.projectType.nextjsComment": "Server-rendered by default. No need to worry about routing. A great choice for quickly getting products built with server-side rendering by a Node.js server.",
  "createNewProjectWizard.summaryPane.projectType.nextjsLearn": "Learn more about Next.js.",
  "createNewProjectWizard.summaryPane.projectIcon.title": "Project Icon",
  "createNewProjectWizard.summaryPane.projectIcon.content": "Choose an icon, to help you recognize this project from a list.",
  "createNewProjectWizard.summaryPane.projectStarter.title": "Gatsby Starter",
  "createNewProjectWizard.summaryPane.projectStarter.enter": "Please enter a starter for your project (e.g. gatsby-starter-blog or repo. url) or pick one from the starters list.",
  "createNewProjectWizard.summaryPane.projectStarter.optional": "This step is optional. Just leave the field empty to use the default Gatsby starter. But picking a starter will help to bootstrap your project e.g. you can easily create your own blog by picking one of the blog starter templates.",
  "createNewProjectWizard.summaryPane.projectStarter.overview": "For a better overview you can also have a look at the ",
  "createNewProjectWizard.summaryPane.projectStarter.library": "Gatsby starters library"
}