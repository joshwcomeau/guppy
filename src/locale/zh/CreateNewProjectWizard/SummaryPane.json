{
  "createNewProjectWizard.summaryPane.start.title": "创建新项目",
  "createNewProjectWizard.summaryPane.start.content": "请先给你的新项目起个名字。",
  "createNewProjectWizard.summaryPane.projectName.title": "项目名称",
  "createNewProjectWizard.summaryPane.projectName.content": "不必太纠结项目名称，之后可以再做修改",
  "createNewProjectWizard.summaryPane.projectType.title": "项目类型",
  "createNewProjectWizard.summaryPane.projectType.content": "Guppy可以创建不同类型的项目。点击一种类型来了解一下。",
  "createNewProjectWizard.summaryPane.projectType.reactIntro": "Vanilla React项目使用create-react-app，这是由Facebook开发的官方命令行工具，用来引导React应用。",
  "createNewProjectWizard.summaryPane.projectType.reactComment": "它是一个很棒的多用途工具，并且是推荐的途径来帮您成为熟练的React开发者。",
  "createNewProjectWizard.summaryPane.projectType.reactLearn": "了解更多关于create-react-app.",
  "createNewProjectWizard.summaryPane.projectType.gatsbyIntro": "Gatsby是一个超快的React静态网页生成器。",
  "createNewProjectWizard.summaryPane.projectType.gatsbyComment": "它很适合用来构建博客和个人网站，并且提供开箱即用的出色性能，是一个很棒的快速完成产品构建的选择。",
  "createNewProjectWizard.summaryPane.projectType.gatsbyLearn": "了解更多关于Gatsby.",
  "createNewProjectWizard.summaryPane.projectType.nextjsIntro": "Next.js是一个用于静态网页或服务端渲染应用的轻量级框架。",
  "createNewProjectWizard.summaryPane.projectType.nextjsComment": "默认使用服务端渲染，无需担心路由，是一个很棒的快速完成Node.js后端渲染产品构建的选择。",
  "createNewProjectWizard.summaryPane.projectType.nextjsLearn": "了解更多关于Next.js.",
  "createNewProjectWizard.summaryPane.projectIcon.title": "项目图标",
  "createNewProjectWizard.summaryPane.projectIcon.content": "选择一个图标以便于从列表中辨识项目。",
  "createNewProjectWizard.summaryPane.projectStarter.title": "Gatsby初始器",
  "createNewProjectWizard.summaryPane.projectStarter.enter": "请输入一个项目初始器（例如gatsby-starter-blog）或从列表中选择一个初始器。",
  "createNewProjectWizard.summaryPane.projectStarter.optional": "这一步是可选的，此栏为空则使用默认Gatsby初始器。但是选择一个初始器将帮助您引导项目，例如选择博客初始模板能帮助您轻松创建属于您的博客。",
  "createNewProjectWizard.summaryPane.projectStarter.overview": "您可以在这里看到一个更好的概述 ",
  "createNewProjectWizard.summaryPane.projectStarter.library": "Gatsby初始器库"
}