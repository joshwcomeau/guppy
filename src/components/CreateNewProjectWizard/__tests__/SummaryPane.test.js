import React from 'react';
import { shallow } from 'enzyme';

import SummaryPane from '../SummaryPane';
import projectTypes from '../../../config/project-types';

describe('SummaryPane component', () => {
  let wrapper;
  const steps = ['projectName', 'projectType', 'projectIcon', 'projectStarter'];

  steps.forEach(step =>
    it(`should render summary for ${step}`, () => {
      if (step === 'projectType') {
        wrapper = shallow(<SummaryPane currentStep={step} />);
        Object.keys(projectTypes).forEach(projectType => {
          wrapper = shallow(
            <SummaryPane currentStep={step} projectType={projectType} />
          );
          expect(wrapper).toMatchSnapshot();
        });
      } else {
        wrapper = shallow(<SummaryPane currentStep={step} />);
        expect(wrapper).toMatchSnapshot();
      }
    })
  );

  it('should throw an error if step not found', () => {
    expect(() =>
      shallow(<SummaryPane currentStep={'not-available'} />)
    ).toThrow();
  });
});
