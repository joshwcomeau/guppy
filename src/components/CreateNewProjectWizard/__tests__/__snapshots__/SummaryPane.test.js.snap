// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryPane component should render summary for projectIcon 1`] = `
<styled.div>
  <styled.div
    key="s3t"
  >
    <styled.h1>
      Project Icon
    </styled.h1>
    <styled.div>
      Choose an icon, to help you recognize this project from a list.
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`SummaryPane component should render summary for projectName 1`] = `
<styled.div>
  <styled.div>
    <styled.div
      key="intro-t"
    >
      <Logo
        size="large"
      />
      <styled.div
        size={30}
      />
      <styled.h1>
        Create new project
      </styled.h1>
      <styled.div>
        Let's start by giving your new project a name.
      </styled.div>
      <styled.div
        size={130}
      />
      <Connect(ImportExisting) />
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`SummaryPane component should render summary for projectStarter 1`] = `
<styled.div>
  <styled.div>
    <styled.h1>
      Gatsby Starter
    </styled.h1>
    <styled.div>
      Please enter a starter for your project (e.g. gatsby-starter-blog or repo. url) or pick one from the starters list.
    </styled.div>
    <styled.div>
      This step is optional. Just leave the field empty to use the default Gatsby starter. But picking a starter will help to bootstrap your project e.g. you can easily create your own blog by picking one of the blog starter templates.
    </styled.div>
    <styled.div>
      For a better overview you can also have a look at the
       
      <ExternalLink
        color="#FFF"
        display="inline-block"
        hoverColor="#FFF"
        href="https://www.gatsbyjs.org/starters/"
      >
        Gatsby starters library
      </ExternalLink>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`SummaryPane component should render summary for projectType 1`] = `
<styled.div>
  <styled.div
    key="s2t"
  >
    <styled.h1>
      Project Type
    </styled.h1>
    <styled.div>
      <strong>
        Vanilla React
      </strong>
    </styled.div>
    <styled.div>
      Vanilla React projects use create-react-app, an official command-line tool built by Facebook for bootstrapping React applications.
    </styled.div>
    <styled.div>
      It's a fantastic general-purpose tool, and is the recommended approach if you're looking to become a skilled React developer.
    </styled.div>
    <styled.div>
      <ExternalLink
        color="#FFF"
        display="inline-block"
        hoverColor="#FFF"
        href="https://github.com/facebook/create-react-app"
      >
        <strong>
          Learn more about create-react-app.
        </strong>
      </ExternalLink>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`SummaryPane component should render summary for projectType 2`] = `
<styled.div>
  <styled.div
    key="s2t"
  >
    <styled.h1>
      Project Type
    </styled.h1>
    <styled.div>
      <strong>
        Gatsby
      </strong>
    </styled.div>
    <styled.div>
      Gatsby is a blazing fast static site generator for React.
    </styled.div>
    <styled.div>
      It's great for building blogs and personal websites, and provides amazing performance out-of-the-box. A great choice for quickly getting products built.
    </styled.div>
    <styled.div>
      <ExternalLink
        color="#FFF"
        display="inline-block"
        hoverColor="#FFF"
        href="https://www.gatsbyjs.org/"
      >
        <strong>
          Learn more about Gatsby.
        </strong>
      </ExternalLink>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`SummaryPane component should render summary for projectType 3`] = `
<styled.div>
  <styled.div
    key="s2t"
  >
    <styled.h1>
      Project Type
    </styled.h1>
    <styled.div>
      <strong>
        Next.js
      </strong>
    </styled.div>
    <styled.div>
      Next.js is a lightweight framework for static and server-rendered applications.
    </styled.div>
    <styled.div>
      Server-rendered by default. No need to worry about routing. A great choice for quickly getting products built with server-side rendering by a Node.js server.
    </styled.div>
    <styled.div>
      <ExternalLink
        color="#FFF"
        display="inline-block"
        hoverColor="#FFF"
        href="https://nextjs.org/learn/"
      >
        <strong>
          Learn more about Next.js.
        </strong>
      </ExternalLink>
    </styled.div>
  </styled.div>
</styled.div>
`;
